go/hack/compat.go

```diff
@@ -0,0 +1,36 @@
+//go:build !gc || wasm
+
+package hack
+
+func String(b []byte) (s string) {
+	return string(b)
+}
+
+func StringBytes(s string) []byte {
+	return []byte(s)
+}
+
+func DisableProtoBufRandomness() {}
+
+func RuntimeMemhash(b []byte, hash uint64) uint64 {
+	for i := 0; i < len(b); i++ {
+		hash *= 1099511628211
```

Comment:
Out of curiosity, what's `1099511628211`?

Reply:
Maybe it has to do with [this](https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function)?

Reply:
The default prime constant in FNV1: https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/hash/fnv/fnv.go;l=36